src/PostgREST/Config.hs

```diff
@@ -373,15 +382,14 @@ parser optPath env dbSettings roleSettings roleIsolationLvl =
         Just "main-query" -> pure  LogQueryMain
         Just _            -> fail "Invalid SQL logging value. Check your configuration."
 
-    parseTxEnd :: C.Key -> ((Bool, Bool) -> Bool) -> C.Parser C.Config Bool
-    parseTxEnd k f =
+    parseTxEnd :: C.Key -> C.Parser C.Config DbTxEnd
+    parseTxEnd k =
       optString k >>= \case
-        --                                          RollbackAll AllowOverride
-        Nothing                        -> pure $ f (False,      False)
-        Just "commit"                  -> pure $ f (False,      False)
-        Just "commit-allow-override"   -> pure $ f (False,      True)
-        Just "rollback"                -> pure $ f (True,       False)
-        Just "rollback-allow-override" -> pure $ f (True,       True)
+        Nothing                        -> pure TxCommit -- default
+        Just "commit"                  -> pure TxCommit
+        Just "commit-allow-override"   -> pure TxCommitAllowOverride
+        Just "rollback"                -> pure TxRollback
+        Just "rollback-allow-override" -> pure TxRollbackAllowOverride
```

Comment:
Also, see how this change is much cleaner and less confusing.

Reply:
Hm, yeah. I always found these booleans a bit confusing.